demo/demo.cpp

```diff
@@ -0,0 +1,152 @@
+/*
+ * This example shows how to use the InOrbit Robot SDK for C++11 to publish
+ * custom key-value data and the robot pose.
+ *
+ * The InOrbit agent must be running before executing this program for it to
+ * work properly.
+ */
+#include "../include/inorbit/inorbit.hpp"
+#include <iostream>
+#include <stdio.h>
+#include <unistd.h>
+
+// The `start` file is used as a signal
+// mechanism to trigger the simulation
+#define DEMO_START_TMP_FILE "/tmp/start"
+#define DEMO_DATA_POINT_DELAY 1
+
+using namespace std;
+
+// Position struct used on Pose
+struct Position {
+  Position(const float &x, const float &y, const float &z)
+    : x(x), y(y), z(z) {}
+  float x, y, z;
+};
+
+// Rotation struct used on Pose
+struct Rotation {
+  Rotation(const float &i, const float &j, const float &k, const float &w)
+    : i(i), j(j), k(k), w(w) {}
+  float i, j, k, w;
+};
+
+// Data point struct used on the simulation
+struct RobotDataPoint {
+  RobotDataPoint(const Position &position, const Rotation &rotation,
+                 const string &status, const float &battery)
+      : position(position), rotation(rotation), status(status), battery(battery) {}
+  Position position;
+  Rotation rotation;
+  string status;
+  float battery;
+};
+
+inline bool file_exists(const std::string& name) {
```

Comment:
It can be called very frequently so it reduces overhead (pushing the func into the stack, copying parameters ...).
Clearly it doesn't matter much for this demo code though